### PR TITLE
CRAFTERCMS-1797

### PR DESCRIPTION
**Pull request by dejan-brkic**
Implemented new Dependency Service

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/1797

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java [hunk 1]
```diff
@@ -657,6 +661,40 @@ protected String copyContent(String site, String fromPath, String toPath, Set<St
         return retNewFileName;
     }
 
+    protected Map<String, String> getCopyDependencies(String site, String sourceContentPath, String dependencyPath) throws ServiceException {
+        Map<String, String> copyDependency = new HashMap<String, String>();
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java:665]: Why not just use getItemSpecificDependencies as an implementation of this method?

File: src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java [hunk 2]
```diff
@@ -808,50 +846,16 @@ protected void updateDatabaseCachePreviewForMove(String site, String fromPath, S
...
-                        dependencyService.extractDependenciesJavascript(site, movePath, assetContent, globalDeps);
-                    } else if (isTemplate) {
-                        dependencyService.extractDependenciesTemplate(site, movePath, assetContent, globalDeps);
-                    }
-                } catch (ServiceException e) {
-                    logger.error("Error while updating dependencies on move content site: " + site + " path: " + movePath, e);
-                }
-            }
+    protected void updateDependenciesOnMove(String site, String fromPath, String movePath) throws ServiceException {
+        dependencyService.deleteItemDependencies(site, fromPath);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java]: Why not just use upsertDependencies which will beginTx, delete, add, commitTx?

File: src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java [hunk 3]
```diff
@@ -2081,7 +2085,7 @@ protected void childDeleteItems(String site, ContentItemTO contentItem, GoLiveDe
     protected void addDependenciesToDelete(String site, String sourceContentPath, String dependencyPath, GoLiveDeleteCandidates candidates,boolean isLiveRepo) throws ServiceException {
         Set<String> dependencyParentFolder = new HashSet<String>();
         //add dependencies as well
-        Set<DmDependencyTO> dmDependencyTOs = dependencyService.getDeleteDependencies(site, sourceContentPath, dependencyPath,isLiveRepo);
+        Set<DmDependencyTO> dmDependencyTOs = getDeleteDependencies(site, sourceContentPath, dependencyPath,isLiveRepo);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java:2088]: Why are we still using DmDependencyTO?

File: src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java [hunk 4]
```diff
@@ -2106,10 +2110,36 @@ protected void addDependenciesToDelete(String site, String sourceContentPath, St
         }
     }
 
+    protected Set<DmDependencyTO> getDeleteDependencies(String site, String sourceContentPath, String dependencyPath, boolean isLiveRepo) throws ServiceException {
+        Set<DmDependencyTO> toRet = new HashSet<DmDependencyTO>();
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java:2114]: Why not just use getItemSpecificDependencies?

File: src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java [hunk 5]
```diff
@@ -2118,6 +2148,45 @@ protected void addRemovedDependenicesToDelete(String site, String relativePath,
         }
     }
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java:2150]: Let's discuss the following protected methods tomorrow.

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/RegexDependencyResolver.java [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+            }
+        } else {
+            String configLocation = getConfigLocation(site);
+            logger.error("Failed to load Dependency Resolver configuration. Verify that configuration exists and it is valid XML file: " + configLocation);
+        }
+        return toRet;
+    }
+
+    @SuppressWarnings("unchecked")
+    private DependencyResolverConfigTO getConfiguraion(String site) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/RegexDependencyResolver.java]: Spelling error: getConfiguration

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,478 @@
...
+            try {
+                logger.debug("Delete all source dependencies for site: " + site + " path: " + path);
+                deleteAllSourceDependencies(site, path);
+                logger.debug("Insert all extracted dependencies entries for site: " + site + " path: " + path);
+                insertDependencies(dependencyEntities);
+                logger.debug("Committing transaction.");
+                transactionManager.commit(txStatus);
+            } catch (Exception e) {
+                logger.debug("Rolling back transaction.");
+                transactionManager.rollback(txStatus);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: Should be in a finally, no?

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,478 @@
...
+                dependencyObj.setType(dependencyType);
+                dependencyEntities.add(dependencyObj);
+                extractedPaths.add(file);
+            }
+        }
+        return dependencyEntities;
+    }
+
+    private String getCleanPath(String path) {
+        String cleanPath = path.replaceAll("//", "/");
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java:109]: Why is this needed?

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,478 @@
...
+        try {
+            logger.debug("Delete all source dependencies for list of paths site: " + site);
+            deleteAllSourceDependencies(site, paths);
+            logger.debug("Insert all extracted dependencies entries lof list of paths for site: " + site);
+            insertDependencies(dependencyEntities);
+            logger.debug("Committing transaction.");
+            transactionManager.commit(txStatus);
+        } catch (Exception e) {
+            logger.debug("Rolling back transaction.");
+            transactionManager.rollback(txStatus);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: Should be in a finally?

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,478 @@
...
+
+    @Override
+    public Set<String> getPublishingDepenencies(String site, String path) throws SiteNotFoundException, ContentNotFoundException, ServiceException {
+        logger.debug("Get publishing dependencies for site: " + site + " path:" + path);
+        List<String> paths = new ArrayList<String>();
+        paths.add(path);
+        return getPublishingDepenencies(site, paths);
+    }
+
+    protected  Set<String> getMandatoryParents(String site, List<String> paths) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: Let's discuss this method.

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Let's review this tomorrow.
I want to review the ContentServiceImpl updates, the new service implementation, and my specific comments.

**Comment by dejan-brkic** [comment]
Added requested changes
